src/file.rs

```diff
@@ -384,7 +384,8 @@ impl Read for File {
                 self.fs,
                 self.f,
                 buf.as_ptr() as *mut c_void,
-                buf.len() as i32,
+                // at most 2^30 bytes, ~1GB
+                buf.len().min(1073741824) as i32,
```

Comment:
Nice. How about adding a `READ_LIMIT` like rust does for file?

Reply:
That would be great. I would like to add a FILE_LIMIT, since this limit apply both to read and write.